vcf-dumper/vcf-dumper-lib/src/main/java/uk/ac/ebi/eva/vcfdump/VariantToVariantContextConverter.java

```diff
@@ -96,57 +97,54 @@ public VariantContext transform(VariantWithSamplesAndAnnotation variant) {
-                        if (soName != null) {
-                            soNameList.add(soName);
-                        }
-                    }
-                }
-                String soNames = String.join("&", soNameList);
-                String symbol = consequenceType.getGeneName();
-                String gene = consequenceType.getEnsemblGeneId();
-                String feature = consequenceType.getEnsemblTranscriptId();
-                String bioType = consequenceType.getBiotype();
-                Integer cDnaPosition = consequenceType.getcDnaPosition();
-                Integer cdsPosition = consequenceType.getCdsPosition();
-
-                StringBuilder csqSb = new StringBuilder();
-                csqSb.append(allele != null ? allele : "").append("|")
-                        .append(soNames != null ? soNames : "").append("|")
-                        .append(symbol != null ? symbol : "").append("|")
-                        .append(gene != null ? gene : "").append("|")
-                        .append(feature != null ? feature : "").append("|")
-                        .append(bioType != null ? bioType : "").append("|")
-                        .append(cDnaPosition != null ? cDnaPosition : "").append("|")
-                        .append(cdsPosition != null ? cdsPosition : "");
-                consequences.add(csqSb.toString());
-            }
-            csq = String.join(",", consequences);
+        return consequenceTypes;
+    }
 
+    private String transformConsequenceTypeToCsqTag(String allele, ConsequenceType consequenceType) {
+        Set<Integer> soAccessions =  consequenceType.getSoAccessions();
+        String soNames = null;
+        if (soAccessions != null) {
+            soNames = soAccessions.stream()
+                    .map(ConsequenceTypeMappings::getSoName).filter(Objects::nonNull)
+                    .collect(Collectors.joining("&"));
         }
-        return csq;
+        String symbol = consequenceType.getGeneName();
+        String gene = consequenceType.getEnsemblGeneId();
+        String feature = consequenceType.getEnsemblTranscriptId();
+        String bioType = consequenceType.getBiotype();
+        Integer cDnaPosition = consequenceType.getcDnaPosition();
+        Integer cdsPosition = consequenceType.getCdsPosition();
+
+        StringBuilder csqSb = new StringBuilder();
+        csqSb.append(allele != null ? allele : "").append("|")
+                .append(soNames != null ? soNames : "").append("|")
+                .append(symbol != null ? symbol : "").append("|")
+                .append(gene != null ? gene : "").append("|")
+                .append(feature != null ? feature : "").append("|")
+                .append(bioType != null ? bioType : "").append("|")
+                .append(cDnaPosition != null ? cDnaPosition : "").append("|")
+                .append(cdsPosition != null ? cdsPosition : "");
```

Comment:
It could be easier to understand if the ternary operators were used in the assignments instead of among all the calls to `append`.